src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -416,11 +416,13 @@ rpmostree_sysroot_upgrader_pull_base (RpmOstreeSysrootUpgrader  *self,
                                      g_variant_new_variant (g_variant_new_string (dir_to_pull)));
             g_variant_builder_add (optbuilder, "{s@v}", "flags",
                                    g_variant_new_variant (g_variant_new_int32 (flags)));
-            /* Add the timestamp check, unless disabled. The option was added in
-             * libostree v2017.11 */
+            /* Add the timestamp check, unless disabled or we're fetching a specific commit
+             * since we have no idea where the ostree ref actually is, and that's what
+             * libostree keys off of. Note in the latter case, we still verify that the
+             * fetched commit is newer than the deployment's base revision below. */
             if (!allow_older)
-              g_variant_builder_add (optbuilder, "{s@v}", "timestamp-check",
-                                     g_variant_new_variant (g_variant_new_boolean (TRUE)));
+              g_variant_builder_add (optbuilder, "{s@v}", "timestamp-check-from-rev",
```

Comment:
This makes rpm-ostree subtly require this API from ostree in a way that breaks downgrade protection if we run with an older libostree.

Reply:
Good catch, thanks! I just bumped the minimum libostree required.